pkg/controller/cassandradatacenter/controller/reconciler.go

```diff
@@ -15,58 +16,66 @@ var log = logf.Log.WithName("CassandraDataCenterReconciler")
 // CassandraDataCenterReconciler reconciles a CassandraDataCenter object
 type CassandraDataCenterReconciler struct {
 	// This client is a split client that reads objects from the cache and writes to the apiserver
-	client client.Client
+	client client.Client // TODO: pointer?
```

Comment:
potentially, I think it was generated like that ...

Reply:
Apparently it already is, because all the methods of all the implementations take pointers.

Reply:
yea, no need to make it a pointer. `client.Client` is an interface, it would make no sense keeping a pointer to an interface.

Reply:
just to be sure we are on the same page, this is not Sidecar client, its some internal client to kubernetes from operator ...